backend/server.js

```diff
@@ -0,0 +1,21 @@
+import { ApolloServer, gql } from "apollo-server"
+import typeDefs from './typeDefs'
+import { Post, InMemoryDataSource } from './db'
+import { resolvers } from './resolvers'
+
+const db = new InMemoryDataSource()
+
+const dataSources = () => ({ db })
+const context = ({ req, res }) => ({ req, res })
```

Comment:
I wouldn't pass `req` and `res` to the datasources and resolvers, only pass what you actually need.